app/forms/base_housing_costs_form.rb

```diff
@@ -35,27 +35,28 @@ def frequencies
 private
 
   def housing_benefit_does_not_exceed_costs
-    # Skip validation if housing benefit isn't relevant or value/frequency is missing
     return unless housing_benefit_relevant &&
-      housing_benefit_value.to_i.positive? &&
+      housing_benefit_value.respond_to?(:positive?) &&
+      housing_benefit_value.positive? &&
       housing_benefit_frequency.present?
 
-    # If housing payments are zero, any housing benefit is excessive
-    if total_annual_housing_costs.zero?
-      errors.add(:housing_benefit_value, :exceeds_costs)
-      return
-    end
+    total_costs = total_annual_housing_costs
+
+    # Return early if total_costs is not a numeric type or is zero
+    return unless total_costs.respond_to?(:zero?) && !total_costs.zero?
```

Comment:
Could this be:

`return unless total_costs.is_a?(Numeric) && total_costs.nonzero?`